modules/xmpp/AVModeration.js

```diff
@@ -0,0 +1,101 @@
+import { $msg } from 'strophe.js';
+
+import * as MediaType from '../../service/RTC/MediaType';
+import XMPPEvents from '../../service/xmpp/XMPPEvents';
+
+/**
+ * The AVModeration logic.
+ */
+export default class AVModeration {
+
+    /**
+     * Constructs lobby room.
+     *
+     * @param {ChatRoom} room the main room.
+     */
+    constructor(room) {
+        this.xmpp = room.xmpp;
+
+        this.mainRoom = room;
+
+        this.enabled = false;
+        this.whitelistAudio = [];
+        this.whitelistVideo = [];
+
+        this.xmpp.addListener(XMPPEvents.AV_MODERATION_RECEIVED, this._onMessage.bind(this));
+    }
+
+    /**
+     * Whether lobby is supported on backend.
+     *
+     * @returns {boolean} whether lobby is supported on backend.
+     */
+    isSupported() {
+        return Boolean(this.xmpp.avModerationComponentAddress);
+    }
+
+    /**
+     * Enables or disabled AV Moderation by sending a msg with command to the component.
+     */
+    enable(state) {
+        if (!this.isSupported() || !this.mainRoom.isModerator()) {
+            return;
+        }
+
+        if (state === this.enabled) {
+            return;
+        }
+
+        // send the enable/disable message
+        const msg = $msg({ to: this.xmpp.avModerationComponentAddress });
+
+        msg.c('av_moderation', { enable: state }).up();
+
+        this.xmpp.connection.send(msg);
+    }
+
+    /**
+     * Enables or disabled AV Moderation by sending a msg with command to the component.
+     */
+    approve(mediaType, jid) {
+        if (!this.isSupported() || !this.mainRoom.isModerator()) {
+            return;
+        }
+
+        // send a message to whitelist the jid and approve it to unmute
+        const msg = $msg({ to: this.xmpp.avModerationComponentAddress });
+
+        msg.c('av_moderation', {
+            mediaType,
+            whitelist: jid }).up();
```

Comment:
I wonder if there's a better name for this, which avoids using `whitelist` as a verb (since we already use it as a noun in other messages). `jidToWhitelist`?

How do you remove someone from the whitelist? Presumably if you're doing a "moderator mute" on someone in the whitelist you want them removed?

Reply:
We do not remove it, no such option right now. You need to disable enable the feature to start over. 
We have discussed that particular case.